File: backend/app/models/api/upload_new.py
# Upload API models for the new v1 upload endpoint

from pydantic import BaseModel, Field
from typing import List, Optional, Any, Union, Dict



class APIUploadRequest(BaseModel):
    """Request model for the new v1 upload endpoint."""
    # This model represents the form data structure sent from frontend
    # The actual files and metadata are handled in the endpoint
    # This is primarily for documentation and validation purposes
    file_count: int = Field(..., description="Total number of files being uploaded")

    # Note: Individual file fields (file_0, file_1, etc.) and their metadata
    # (document_type_0, file_type_0, etc.) are handled directly in the endpoint
    # as they come from FormData


class APIUploadResponse(BaseModel):
    """Response model for the upload API endpoint."""
    success: bool = Field(..., description="Whether the upload was successful")
    message: str = Field(..., description="Response message")
    deal_id: str = Field(..., description="ID of the created deal")
    files: List[Dict[str, str]] = Field(..., description="List of uploaded files with metadata")
    om_file_url: Optional[str] = Field(None, description="Signed URL to the OM PDF file")
    t12_file_url: Optional[str] = Field(None, description="Signed URL to the T12 file (if available)")
    rent_roll_file_url: Optional[str] = Field(None, description="Signed URL to the rent roll file (if available)")
    property_info: Optional[Dict[str, Any]] = Field(None, description="Property information from OM classification")
    structured_t12: Optional[Union[List[Any], Dict[str, Any]]] = Field(None, description="Structured T12 data")
    structured_rent_roll: Optional[Union[List[Any], Dict[str, Any]]] = Field(None, description="Structured rent roll data")
    classification_result: Optional[Dict[str, Any]] = Field(None, description="OM classification result")
    error_code: Optional[str] = Field(None, description="Error code if upload failed")
    details: Optional[Dict[str, Any]] = Field(None, description="Additional error details")

File: frontend/types/api/uploadNew.ts
// Upload API types that exactly match the backend models

import { EnhancedClassificationResult } from '@/types/property';

// APIUploadRequest - matches backend APIUploadRequest
export interface APIUploadRequest {
  file_count: number;
  // Note: Individual file fields (file_0, file_1, etc.) and their metadata
  // (document_type_0, file_type_0, etc.) are handled directly in the endpoint
  // as they come from FormData
}

// APIUploadResponse - matches backend APIUploadResponse
export interface APIUploadResponse {
  success: boolean;
  message: string;

  // Deal and file information
  deal_id: string;
  files: Array<{
    filename: string;
    document_type: string;
    file_type: string;
    file_url: string;
  }>;

  // File URLs for different document types
  om_file_url?: string | null;
  t12_file_url?: string | null;
  rent_roll_file_url?: string | null;

  // Property information from OM classification
  property_info?: Record<string, any> | null;

  // Structured outputs
  structured_t12?: Array<Record<string, any>> | null;
  structured_rent_roll?: Array<Record<string, any>> | null;

  // Raw extraction results (for debugging/advanced use cases)
  classification_result?: EnhancedClassificationResult | null;

  // Error information
  error_code?: string | null;
  details?: Record<string, any> | null;
}
